lib/graphql/query/arguments.rb

```diff
@@ -10,8 +10,14 @@ class Arguments
       def initialize(values, argument_definitions:)
         @original_values = values
         @argument_values = values.inject({}) do |memo, (inner_key, inner_value)|
-          string_key = inner_key.to_s
-          arg_defn = argument_definitions[string_key]
+          arg_defn = argument_definitions[inner_key.to_s]
+
+          string_key = if arg_defn.as != nil
+            arg_defn.as.to_s
```

Comment:
Do we need to `to_s` this at runtime, or could we do it during definition, like 

```
argument.as = rest[:as].to_s
```

?